src/testdrive/src/action/kafka/verify.rs

```diff
@@ -165,10 +183,25 @@ impl Action for VerifyAction {
                     consumer
                         .store_offset_from_message(&message)
                         .context("storing message offset")?;
-                    actual_bytes.push((
-                        message.key().and_then(|bytes| Some(bytes.to_owned())),
-                        message.payload().and_then(|bytes| Some(bytes.to_owned())),
-                    ));
+                    let headers = self
+                        .header_keys
+                        .iter()
+                        .map(|k| {
+                            // grab the value of the first header with the matching key, if there is one.
+                            message
+                                .headers()
+                                .and_then(|h| h.iter().find(|i| i.key == k))
+                                .and_then(|h| h.value)
+                                .and_then(|v| str::from_utf8(v).ok())
+                                .map(str::to_owned)
+                                .unwrap_or_else(|| "".to_string())
+                        })
+                        .collect();
+                    actual_bytes.push(Record {
+                        headers,
+                        key: message.key().and_then(|bytes| Some(bytes.to_owned())),
```

Comment:
nit: `message.key().map(|bytes| bytes.to_owned())` or `message.key().map(ToOwned::to_owned)` are a little more idiomatic I think

Reply:
oof, yeah, not sure where i was going with this one...

Reply:
ah right, this is from the old implementation. might as well tidy up while we're here!